pkg/gitops.go

```diff
@@ -104,17 +111,27 @@ func CommitAndPush(outputPath string, repoPathOverride string, remote string, co
 	return nil
 }
 
-func GenerateAndPush(outputPath string, remote string, component gitopsv1alpha1.Component, e Executor, appFs afero.Afero, branch string, doPush bool, createdBy string, commonStorage *corev1.PersistentVolumeClaim) error {
+// GenerateAndPush generates a new gitops folder with one component, and optionally pushes to Git. Note: this does not
+// clone an existing gitops repo.
+// 1. outputPath: Where the gitops resources are
+// 2. remote: A string of the form https://$token@github.com/<org>/<repo>. Corresponds to the component's gitops repository
+// 3. component: The component corresponding to a single Component in an Application in AS. eg. component.Name
+// 4. The executor to use to execute the git commands (either gitops.executor or gitops.mockExecutor)
+// 5. The filesystem object used to create (either ioutils.NewFilesystem() or ioutils.NewMemoryFilesystem())
+// 6. The branch to push to
+// 7. Optionally push to the GitOps repository or not.  Default is not to push.
+// 8. createdBy: Use a unique name to identify that clients are generating the GitOps repository. Default is "application-service" and should be overwritten.
+func GenerateAndPush(outputPath string, remote string, component gitopsv1alpha1.GeneratorOptions, e Executor, appFs afero.Afero, branch string, doPush bool, createdBy string) error {
 	CreatedBy = createdBy
 
-	componentName := component.Spec.ComponentName
-	repoPath := filepath.Join(outputPath, component.Spec.Application)
+	componentName := component.Name
+	repoPath := filepath.Join(outputPath, component.Application)
 
 	// Generate the gitops resources and update the parent kustomize yaml file
 	gitopsFolder := repoPath
 
-	gitHostAccessToken := component.Spec.Secret
-	gitOpsRepoURL := component.Spec.Source.GitSource.URL
+	gitHostAccessToken := component.Secret
+	gitOpsRepoURL := component.GitSource.URL
```

Comment:
`component.GitSource` will panic if its nil

Reply:
Thanks for catching this.  I also moved it into the `if doPush` block just below where it is actually used.